[CORL-664] Threshold moderate new commenters

nick-funk opened the pull request:
## What does this PR do?

This is the same changes @tessalt previously wrote with two minor changes.

- Updates the old branch to latest master with all the fragment and style changes around the moderation/config area
- Replaces the prior counting calculation with a simple lookup from the user's cached `commentCounts` introduced in CORL-688

If there is a `don't merge` label on this PR, it's due to the fact that `CORL-688` likely hasn't been merged yet.

## How do I test this PR?

- Login as an admin/mod
- Head to the Moderation config and enable new commenter approval feature
- Default approves is likely set to 2
- Create a new commenter and sign in
- Make a couple comments on the stream
- See that they get sent for moderation
- Check back in the admin side with your mod account in the pending queue
- Approve 2 of their comments (or whatever your approval threshold setting was)
- Go back to the stream and post a comment as the commenter
- Their comments should be showing up in stream and not be held for moderation

NOTE: make sure you turn auto pre-mod off on the stream you choose to test with

tessalt committed:
add new commenters config

tessalt committed:
fix specs and fixtures, add translation strings

tessalt committed:
save whether a commenter is new

tessalt committed:
fix specs and snaps

tessalt committed:
add admin role to new config options

tessalt committed:
Update copy

tessalt committed:
remvoe unused ref

wyattjoh committed:
feat: initial impl

nick-funk committed:
Create preliminary comment moderation slices

CORL-688

nick-funk committed:
Move slices logic into stacks

CORL-688

nick-funk committed:
Create user comment counts

CORL-688

nick-funk committed:
Create naive mutation that initializes user comment counts

CORL-688

nick-funk committed:
Use bulk updates in user counts migration

CORL-688

wyattjoh committed:
fix: review

wyattjoh committed:
fix: fixed issue with aggregation

nick-funk committed:
Migrate creating comment into stacks

CORL-688

nick-funk committed:
Migrate editing a comment to the stacks

CORL-688

nick-funk committed:
Break publishing comment status out of updateAllCounts

CORL-688

nick-funk committed:
Merge remote-tracking branch 'origin/master' into feature/CORL-664-new-commenters

# Conflicts:
#	src/core/client/admin/components/ModerateCard/MarkersContainer.spec.tsx
#	src/core/client/admin/components/ModerateCard/MarkersContainer.tsx
#	src/core/client/admin/routes/Configure/sections/Moderation/ModerationConfig.tsx
#	src/core/client/admin/routes/Configure/sections/Moderation/ModerationConfigContainer.tsx
#	src/core/client/admin/routes/Configure/sections/Moderation/__snapshots__/ModerationConfig.spec.tsx.snap
#	src/core/client/admin/test/configure/__snapshots__/moderation.spec.tsx.snap
#	src/core/client/admin/test/fixtures.ts
#	src/core/client/test/helpers/fixture.ts
#	src/core/server/graph/tenant/schema/schema.graphql
#	src/locales/en-US/admin.ftl

nick-funk committed:
Merge remote-tracking branch 'origin/feature/CORL-688' into feature/CORL-664-new-commenters

# Conflicts:
#	src/core/server/models/user/user.ts

wyattjoh committed:
Merge branch 'master' into feature/CORL-688

wyattjoh committed:
Merge branch 'master' into feature/CORL-688

wyattjoh committed:
Merge branch 'master' into feature/CORL-688

wyattjoh committed:
review: removed variable scoping in favor of export

wyattjoh committed:
review: removed variable scoping in favor of export

wyattjoh committed:
revert: feb8e8196cd448f5cd24f1ca2eb0b91fe9bd43c7

wyattjoh committed:
revert: feb8e8196cd448f5cd24f1ca2eb0b91fe9bd43c7

wyattjoh commented on src/core/server/models/user/user.ts:
```diff
@@ -471,6 +471,17 @@ export interface User extends TenantResource {
+   * comments than the new commenter moderation config allows to
+   * be considered no longer a new user.
+   */
+  isNew: boolean;
```
This cannot be stored on the user model document as it is dynamically determined by inspecting the counts on a user.

wyattjoh commented on src/core/server/models/user/user.ts:
```diff
@@ -2460,3 +2472,31 @@ export async function deleteModeratorNote(
+  }
+
+  return result.value;
+}
```
This is not required when computing dynamically.

wyattjoh commented on src/core/server/services/comments/pipeline/phases/premodNewCommenter.ts:
```diff
@@ -0,0 +1,50 @@
+
+  if (!author.isNew) {
+    return;
+  }
```
Should be removed.

wyattjoh commented on src/core/server/services/comments/pipeline/phases/premodNewCommenter.ts:
```diff
@@ -0,0 +1,50 @@
+  } else {
+    await markUserNotNew(mongo, tenant.id, author.id);
+  }
+};
```
Should be removed.

wyattjoh commented on src/core/server/models/user/user.ts:
```diff
@@ -525,6 +536,7 @@ async function findOrCreateUserInput(
     moderatorNotes: [],
     digests: [],
     createdAt: now,
+    isNew: true,
```
Should be removed.

wyattjoh commented on src/core/server/graph/tenant/schema/schema.graphql:
```diff
@@ -1921,6 +1945,12 @@ type User {
+  isNew is whether the commenter is a new user as defined
+  by the new commenter moderation rules (if enabled).
+  """
+  isNew: Boolean!
```
Should be removed.

wyattjoh committed:
review: simplification of stacks implementation

This simplifies the stacks implementation to better reuse code related
to count management and event publishing. This can be used to great
effect with the upcomming events PR #2738.

wyattjoh committed:
review: simplification of stacks implementation

This simplifies the stacks implementation to better reuse code related
to count management and event publishing. This can be used to great
effect with the upcomming events PR #2738.

nick-funk committed:
Merge remote-tracking branch 'origin/feature/CORL-688' into feature/CORL-664-new-commenters

# Conflicts:
#	src/core/server/models/user/user.ts

nick-funk committed:
Remove un-necessary isNew flags on users

CORL-664

nick-funk committed:
fix: check if authorID is null before update user counts

CORL-688

nick-funk committed:
Merge remote-tracking branch 'origin/feature/CORL-688' into feature/CORL-664-new-commenters

# Conflicts:
#	src/core/server/stacks/helpers/updateAllCounts.ts

nick-funk committed:
Merge remote-tracking branch 'origin/master' into feature/CORL-664-new-commenters

# Conflicts:
#	src/core/server/models/user/user.ts

wyattjoh committed:
fix: addressed bug in shared count retrival
